event.cpp

```diff
@@ -529,23 +529,23 @@ void cXMLTVEvent::GetSQL(const char *Source, int SrcIdx, const char *ChannelID,
     }
 
     std::string si=sql_insert;
-    int ireps;
-    ireps=pcrecpp::RE("'").GlobalReplace("''",&si);
-    ireps+=pcrecpp::RE("\\^").GlobalReplace("'",&si);
-    ireps+=pcrecpp::RE("'NULL'").GlobalReplace("NULL",&si);
-    if (ireps)
+    long unsigned int ilen = si.length();
+    std::regex_replace(si, std::regex("'"), "''");
+    std::regex_replace(si, std::regex("\\^"), "'");
+    std::regex_replace(si, std::regex("'NULL'"), "NULL");
+    if (ilen != si.length())
     {
```

Comment:
I have no clue what this code actually should do as I never did RegEx in C(++) but let's give it a try and maybe you can correct me:

1. replaces every occurence of a single quote (') with two single quotes ('') (doubles up the "'" character).
2. replaces "^" with a single quote ('). We have to keep in mind that the escaping, done here, is handled by the RegEx library as (^) is a control character in RegEx, if not excaped. So it actually just replaces "^" and we can ignore the "\\".
3. replaces ('NULL') with (NULL) (removes single quotes).

Maybe I'm totally off here, but couldn't this be all done with a simple string replacement? Why RegEx?

BUT there probably is an issue with your "was something changed" check. The first replacement increases length, the second and third one decreases length. So it would be possible to get into a case where this cancels out and some replacements were made but not caught by the "if" statement.

Maybe the only reason, why RegEx was used here, is to know if changes were made. Going over length is a possibility but only if the potential issue above is fixed.